diff --git a/src/main/java/org/apache/commons/crypto/cipher/Cipher.java b/src/main/java/org/apache/commons/crypto/cipher/Cipher.java
index 3cbbe3a..8ba2380 100644
--- a/src/main/java/org/apache/commons/crypto/cipher/Cipher.java
+++ b/src/main/java/org/apache/commons/crypto/cipher/Cipher.java
@@ -38,13 +38,13 @@ public interface Cipher extends Closeable {
    * A constant representing encrypt mode.  The mode constant to be used
    * when calling init method of the Cipher.
    */
-  int ENCRYPT_MODE = javax.crypto.Cipher.ENCRYPT_MODE;
+  int ENCRYPT_MODE = 1;
 
   /**
    * A constant representing decrypt mode.  The mode constant to be used 
    * when calling init method of the Cipher.
    */
-  int DECRYPT_MODE = javax.crypto.Cipher.DECRYPT_MODE;
+  int DECRYPT_MODE = 0;
 
   /**
    * Gets the CipherTransformation for this cipher.
