diff --git a/src/main/java/org/apache/commons/crypto/random/CryptoRandomFactory.java b/src/main/java/org/apache/commons/crypto/random/CryptoRandomFactory.java
index b299353..e9b7a55 100644
--- a/src/main/java/org/apache/commons/crypto/random/CryptoRandomFactory.java
+++ b/src/main/java/org/apache/commons/crypto/random/CryptoRandomFactory.java
@@ -40,8 +40,8 @@ public class CryptoRandomFactory {
 
     /**
      * Gets a CryptoRandom instance for specified props.
-     * Uses the COMMONS_CRYPTO_SECURE_RANDOM_CLASSES_KEY from the provided 
-     * properties.
+     * Uses the {@link #COMMONS_CRYPTO_SECURE_RANDOM_CLASSES_KEY}
+     * from the provided properties.
      * If it is not set, then it checks the System properties.
      * Failing that, it defaults to {@link JavaCryptoRandom}
      * The properties are passed to the generated class.
