diff --git a/src/main/java/org/apache/commons/crypto/stream/CTRCryptoInputStream.java b/src/main/java/org/apache/commons/crypto/stream/CTRCryptoInputStream.java
index 50de2f6..bfdd6fb 100644
--- a/src/main/java/org/apache/commons/crypto/stream/CTRCryptoInputStream.java
+++ b/src/main/java/org/apache/commons/crypto/stream/CTRCryptoInputStream.java
@@ -119,7 +119,7 @@ public class CTRCryptoInputStream extends CryptoInputStream {
      * @param iv Initialization vector for the cipher.
      * @throws IOException if an I/O error occurs.
      */
-    protected CTRCryptoInputStream(InputStream in, CryptoCipher cipher,
+    public CTRCryptoInputStream(InputStream in, CryptoCipher cipher,
             int bufferSize, byte[] key, byte[] iv) throws IOException {
         this(in, cipher, bufferSize, key, iv, 0);
     }
@@ -134,7 +134,7 @@ public class CTRCryptoInputStream extends CryptoInputStream {
      * @param iv Initialization vector for the cipher.
      * @throws IOException if an I/O error occurs.
      */
-    protected CTRCryptoInputStream(ReadableByteChannel in, CryptoCipher cipher,
+    public CTRCryptoInputStream(ReadableByteChannel in, CryptoCipher cipher,
             int bufferSize, byte[] key, byte[] iv) throws IOException {
         this(in, cipher, bufferSize, key, iv, 0);
     }
@@ -149,7 +149,7 @@ public class CTRCryptoInputStream extends CryptoInputStream {
      * @param iv Initialization vector for the cipher.
      * @throws IOException if an I/O error occurs.
      */
-    protected CTRCryptoInputStream(Input input, CryptoCipher cipher,
+    public CTRCryptoInputStream(Input input, CryptoCipher cipher,
             int bufferSize, byte[] key, byte[] iv) throws IOException {
         this(input, cipher, bufferSize, key, iv, 0);
     }
@@ -201,7 +201,7 @@ public class CTRCryptoInputStream extends CryptoInputStream {
      * @param streamOffset the start offset in the stream.
      * @throws IOException if an I/O error occurs.
      */
-    protected CTRCryptoInputStream(InputStream in, CryptoCipher cipher,
+    public CTRCryptoInputStream(InputStream in, CryptoCipher cipher,
             int bufferSize, byte[] key, byte[] iv, long streamOffset)
             throws IOException {
         this(new StreamInput(in, bufferSize), cipher, bufferSize, key, iv,
@@ -219,7 +219,7 @@ public class CTRCryptoInputStream extends CryptoInputStream {
      * @param streamOffset the start offset in the stream.
      * @throws IOException if an I/O error occurs.
      */
-    protected CTRCryptoInputStream(ReadableByteChannel in, CryptoCipher cipher,
+    public CTRCryptoInputStream(ReadableByteChannel in, CryptoCipher cipher,
             int bufferSize, byte[] key, byte[] iv, long streamOffset)
             throws IOException {
         this(new ChannelInput(in), cipher, bufferSize, key, iv, streamOffset);
@@ -236,7 +236,7 @@ public class CTRCryptoInputStream extends CryptoInputStream {
      * @param streamOffset the start offset in the stream.
      * @throws IOException if an I/O error occurs.
      */
-    protected CTRCryptoInputStream(Input input, CryptoCipher cipher,
+    public CTRCryptoInputStream(Input input, CryptoCipher cipher,
             int bufferSize, byte[] key, byte[] iv, long streamOffset)
             throws IOException {
         super(input, cipher, bufferSize, new SecretKeySpec(key, "AES"),
diff --git a/src/main/java/org/apache/commons/crypto/stream/CTRCryptoOutputStream.java b/src/main/java/org/apache/commons/crypto/stream/CTRCryptoOutputStream.java
index 4a25c95..2b86c6d 100644
--- a/src/main/java/org/apache/commons/crypto/stream/CTRCryptoOutputStream.java
+++ b/src/main/java/org/apache/commons/crypto/stream/CTRCryptoOutputStream.java
@@ -119,7 +119,7 @@ public class CTRCryptoOutputStream extends CryptoOutputStream {
      * @param iv Initialization vector for the cipher.
      * @throws IOException if an I/O error occurs.
      */
-    protected CTRCryptoOutputStream(OutputStream out, CryptoCipher cipher,
+    public CTRCryptoOutputStream(OutputStream out, CryptoCipher cipher,
             int bufferSize, byte[] key, byte[] iv) throws IOException {
         this(out, cipher, bufferSize, key, iv, 0);
     }
@@ -134,7 +134,7 @@ public class CTRCryptoOutputStream extends CryptoOutputStream {
      * @param iv Initialization vector for the cipher.
      * @throws IOException if an I/O error occurs.
      */
-    protected CTRCryptoOutputStream(WritableByteChannel channel,
+    public CTRCryptoOutputStream(WritableByteChannel channel,
             CryptoCipher cipher, int bufferSize, byte[] key, byte[] iv)
             throws IOException {
         this(channel, cipher, bufferSize, key, iv, 0);
@@ -150,7 +150,7 @@ public class CTRCryptoOutputStream extends CryptoOutputStream {
      * @param iv Initialization vector for the cipher.
      * @throws IOException if an I/O error occurs.
      */
-    protected CTRCryptoOutputStream(Output output, CryptoCipher cipher,
+    public CTRCryptoOutputStream(Output output, CryptoCipher cipher,
             int bufferSize, byte[] key, byte[] iv) throws IOException {
         this(output, cipher, bufferSize, key, iv, 0);
     }
@@ -202,7 +202,7 @@ public class CTRCryptoOutputStream extends CryptoOutputStream {
      * @param streamOffset the start offset in the data.
      * @throws IOException if an I/O error occurs.
      */
-    protected CTRCryptoOutputStream(OutputStream out, CryptoCipher cipher,
+    public CTRCryptoOutputStream(OutputStream out, CryptoCipher cipher,
             int bufferSize, byte[] key, byte[] iv, long streamOffset)
             throws IOException {
         this(new StreamOutput(out, bufferSize), cipher, bufferSize, key, iv,
@@ -220,7 +220,7 @@ public class CTRCryptoOutputStream extends CryptoOutputStream {
      * @param streamOffset the start offset in the data.
      * @throws IOException if an I/O error occurs.
      */
-    protected CTRCryptoOutputStream(WritableByteChannel channel,
+    public CTRCryptoOutputStream(WritableByteChannel channel,
             CryptoCipher cipher, int bufferSize, byte[] key, byte[] iv,
             long streamOffset) throws IOException {
         this(new ChannelOutput(channel), cipher, bufferSize, key, iv,
@@ -238,7 +238,7 @@ public class CTRCryptoOutputStream extends CryptoOutputStream {
      * @param streamOffset the start offset in the data.
      * @throws IOException if an I/O error occurs.
      */
-    protected CTRCryptoOutputStream(Output output, CryptoCipher cipher,
+    public CTRCryptoOutputStream(Output output, CryptoCipher cipher,
             int bufferSize, byte[] key, byte[] iv, long streamOffset)
             throws IOException {
         super(output, cipher, bufferSize, new SecretKeySpec(key, "AES"),
diff --git a/src/main/java/org/apache/commons/crypto/stream/CryptoInputStream.java b/src/main/java/org/apache/commons/crypto/stream/CryptoInputStream.java
index 59a719b..86fa891 100644
--- a/src/main/java/org/apache/commons/crypto/stream/CryptoInputStream.java
+++ b/src/main/java/org/apache/commons/crypto/stream/CryptoInputStream.java
@@ -130,7 +130,7 @@ public class CryptoInputStream extends InputStream implements
      * @param params the algorithm parameters.
      * @throws IOException if an I/O error occurs.
      */
-    protected CryptoInputStream(InputStream in, CryptoCipher cipher,
+    public CryptoInputStream(InputStream in, CryptoCipher cipher,
             int bufferSize, Key key, AlgorithmParameterSpec params)
             throws IOException {
         this(new StreamInput(in, bufferSize), cipher, bufferSize, key, params);
@@ -146,7 +146,7 @@ public class CryptoInputStream extends InputStream implements
      * @param params the algorithm parameters.
      * @throws IOException if an I/O error occurs.
      */
-    protected CryptoInputStream(ReadableByteChannel in, CryptoCipher cipher,
+    public CryptoInputStream(ReadableByteChannel in, CryptoCipher cipher,
             int bufferSize, Key key, AlgorithmParameterSpec params)
             throws IOException {
         this(new ChannelInput(in), cipher, bufferSize, key, params);
@@ -162,7 +162,7 @@ public class CryptoInputStream extends InputStream implements
      * @param params the algorithm parameters.
      * @throws IOException if an I/O error occurs.
      */
-    protected CryptoInputStream(Input input, CryptoCipher cipher, int bufferSize,
+    public CryptoInputStream(Input input, CryptoCipher cipher, int bufferSize,
             Key key, AlgorithmParameterSpec params) throws IOException {
         this.input = input;
         this.cipher = cipher;
diff --git a/src/main/java/org/apache/commons/crypto/stream/CryptoOutputStream.java b/src/main/java/org/apache/commons/crypto/stream/CryptoOutputStream.java
index 01bd691..f5f9d97 100644
--- a/src/main/java/org/apache/commons/crypto/stream/CryptoOutputStream.java
+++ b/src/main/java/org/apache/commons/crypto/stream/CryptoOutputStream.java
@@ -129,7 +129,7 @@ public class CryptoOutputStream extends OutputStream implements
      * @param params the algorithm parameters.
      * @throws IOException if an I/O error occurs.
      */
-    protected CryptoOutputStream(OutputStream out, CryptoCipher cipher,
+    public CryptoOutputStream(OutputStream out, CryptoCipher cipher,
             int bufferSize, Key key, AlgorithmParameterSpec params)
             throws IOException {
         this(new StreamOutput(out, bufferSize), cipher, bufferSize, key, params);
@@ -145,7 +145,7 @@ public class CryptoOutputStream extends OutputStream implements
      * @param params the algorithm parameters.
      * @throws IOException if an I/O error occurs.
      */
-    protected CryptoOutputStream(WritableByteChannel channel, CryptoCipher cipher,
+    public CryptoOutputStream(WritableByteChannel channel, CryptoCipher cipher,
             int bufferSize, Key key, AlgorithmParameterSpec params)
             throws IOException {
         this(new ChannelOutput(channel), cipher, bufferSize, key, params);
diff --git a/src/main/java/org/apache/commons/crypto/stream/PositionedCryptoInputStream.java b/src/main/java/org/apache/commons/crypto/stream/PositionedCryptoInputStream.java
index 3f0ee76..c7399dd 100644
--- a/src/main/java/org/apache/commons/crypto/stream/PositionedCryptoInputStream.java
+++ b/src/main/java/org/apache/commons/crypto/stream/PositionedCryptoInputStream.java
@@ -84,7 +84,7 @@ public class PositionedCryptoInputStream extends CTRCryptoInputStream {
      * @param streamOffset the start offset in the data.
      * @throws IOException if an I/O error occurs.
      */
-    protected PositionedCryptoInputStream(Input input, CryptoCipher cipher,
+    public PositionedCryptoInputStream(Input input, CryptoCipher cipher,
             int bufferSize, byte[] key, byte[] iv, long streamOffset)
             throws IOException {
         super(input, cipher, bufferSize, key, iv, streamOffset);
