diff --git a/src/main/java/org/apache/commons/crypto/cipher/JceCipher.java b/src/main/java/org/apache/commons/crypto/cipher/JceCipher.java
index e39ef0a..2098b43 100644
--- a/src/main/java/org/apache/commons/crypto/cipher/JceCipher.java
+++ b/src/main/java/org/apache/commons/crypto/cipher/JceCipher.java
@@ -88,8 +88,11 @@ public class JceCipher implements CryptoCipher {
         Utils.checkNotNull(key);
         Utils.checkNotNull(params);
 
-        // Jce uses the javax.crypto.Cipher modes; no need to convert the input
-        cipher.init(mode, key, params);
+        int cipherMode = Cipher.DECRYPT_MODE;
+        if (mode == Cipher.ENCRYPT_MODE) {
+            cipherMode = Cipher.ENCRYPT_MODE;
+        }
+        cipher.init(cipherMode, key, params);
     }
 
     /**
