diff --git a/src/main/java/org/apache/commons/crypto/cipher/CipherFactory.java b/src/main/java/org/apache/commons/crypto/cipher/CipherFactory.java
index fdaf64c..fb712d2 100644
--- a/src/main/java/org/apache/commons/crypto/cipher/CipherFactory.java
+++ b/src/main/java/org/apache/commons/crypto/cipher/CipherFactory.java
@@ -79,15 +79,17 @@ public class CipherFactory {
    */
   public static Cipher getInstance(CipherTransformation transformation)
       throws GeneralSecurityException {
-    return getInstance(transformation, new Properties());
+    return getInstance(transformation, null);
   }
 
-  // Return OpenSSLCipher if Properties is null or empty by default
   private static List<Class<? extends Cipher>> getCipherClasses(Properties props) {
     List<Class<? extends Cipher>> result = new ArrayList<Class<? extends
         Cipher>>();
     String cipherClassString = Utils.getCipherClassString(props);
-
+    if (cipherClassString == null) {
+      LOG.debug("No cipher classes configured.");
+      return null;
+    }
     for (String c : Utils.splitClassNames(cipherClassString, ",")) {
       try {
         Class<?> cls = ReflectionUtils.getClassByName(c);
diff --git a/src/main/java/org/apache/commons/crypto/conf/ConfigurationKeys.java b/src/main/java/org/apache/commons/crypto/conf/ConfigurationKeys.java
index ae07758..494c28d 100644
--- a/src/main/java/org/apache/commons/crypto/conf/ConfigurationKeys.java
+++ b/src/main/java/org/apache/commons/crypto/conf/ConfigurationKeys.java
@@ -17,7 +17,7 @@
  */
 package org.apache.commons.crypto.conf;
 
-import org.apache.commons.crypto.cipher.OpensslCipher;
+import org.apache.commons.crypto.cipher.JceCipher;
 
 /**
  * The ConfigurationKeys contains Configuration keys and default values.
@@ -53,7 +53,7 @@ public class ConfigurationKeys {
    * The default value for crypto cipher.
    */
   public static final String COMMONS_CRYPTO_CIPHER_CLASSES_DEFAULT =
-      OpensslCipher.class.getName();
+      JceCipher.class.getName();
 
   /**
    * The configuration key of the provider class for JCE cipher.
diff --git a/src/main/java/org/apache/commons/crypto/utils/Utils.java b/src/main/java/org/apache/commons/crypto/utils/Utils.java
index 3c2ef86..ec2a1e4 100644
--- a/src/main/java/org/apache/commons/crypto/utils/Utils.java
+++ b/src/main/java/org/apache/commons/crypto/utils/Utils.java
@@ -138,13 +138,8 @@ public class Utils {
    */
   public static String getCipherClassString(Properties props) {
     final String configName = COMMONS_CRYPTO_CIPHER_CLASSES_KEY;
-    String cipherClassString = props.getProperty(configName) != null ? props
-        .getProperty(configName, COMMONS_CRYPTO_CIPHER_CLASSES_DEFAULT) : System
+    return props.getProperty(configName) != null ? props.getProperty(configName) : System
         .getProperty(configName, COMMONS_CRYPTO_CIPHER_CLASSES_DEFAULT);
-    if (cipherClassString.isEmpty()) {
-      cipherClassString = COMMONS_CRYPTO_CIPHER_CLASSES_DEFAULT;
-    }
-    return cipherClassString;
   }
 
   /**
