diff --git a/src/main/java/org/apache/commons/crypto/random/OsCryptoRandom.java b/src/main/java/org/apache/commons/crypto/random/OsCryptoRandom.java
index d105e56..1525023 100644
--- a/src/main/java/org/apache/commons/crypto/random/OsCryptoRandom.java
+++ b/src/main/java/org/apache/commons/crypto/random/OsCryptoRandom.java
@@ -23,8 +23,8 @@ import java.io.IOException;
 import java.util.Properties;
 import java.util.Random;
 
-import org.apache.commons.crypto.conf.ConfigurationKeys;
 import org.apache.commons.crypto.utils.IOUtils;
+import org.apache.commons.crypto.utils.Utils;
 
 /**
  * A Random implementation that uses random bytes sourced from the operating
@@ -66,7 +66,7 @@ public class OsCryptoRandom extends Random implements CryptoRandom {
      * @param props the configuration properties.
      */
     public OsCryptoRandom(Properties props) {
-        randomDevPath = getRandomDevPath(props);
+        randomDevPath = Utils.getRandomDevPath(props);
         File randomDevFile = new File(randomDevPath);
 
         try {
@@ -132,23 +132,4 @@ public class OsCryptoRandom extends Random implements CryptoRandom {
             stream = null;
         }
     }
-
-    /**
-     * Gets the random device path.
-     *
-     * @param props The <code>Properties</code> class represents a set of
-     *        properties.
-     * @return the random device path based on the props.
-     */
-    private static String getRandomDevPath(Properties props) {
-        String devPath = props
-            .getProperty(ConfigurationKeys.COMMONS_CRYPTO_SECURE_RANDOM_DEVICE_FILE_PATH_KEY);
-        if (devPath == null) {
-            devPath = System.getProperty(
-                ConfigurationKeys.COMMONS_CRYPTO_SECURE_RANDOM_DEVICE_FILE_PATH_KEY,
-                ConfigurationKeys.COMMONS_CRYPTO_SECURE_RANDOM_DEVICE_FILE_PATH_DEFAULT);
-        }
-        return devPath;
-    }
-
 }
diff --git a/src/main/java/org/apache/commons/crypto/utils/Utils.java b/src/main/java/org/apache/commons/crypto/utils/Utils.java
index 83a785f..8886377 100644
--- a/src/main/java/org/apache/commons/crypto/utils/Utils.java
+++ b/src/main/java/org/apache/commons/crypto/utils/Utils.java
@@ -157,6 +157,24 @@ public final class Utils {
         return cipherClassString;
     }
 
+    /**
+     * Gets the random device path.
+     *
+     * @param props The <code>Properties</code> class represents a set of
+     *        properties.
+     * @return the random device path based on the props.
+     */
+    public static String getRandomDevPath(Properties props) {
+        String devPath = props
+                .getProperty(ConfigurationKeys.COMMONS_CRYPTO_SECURE_RANDOM_DEVICE_FILE_PATH_KEY);
+        if (devPath == null) {
+            devPath = System.getProperty(
+                    ConfigurationKeys.COMMONS_CRYPTO_SECURE_RANDOM_DEVICE_FILE_PATH_KEY,
+                    ConfigurationKeys.COMMONS_CRYPTO_SECURE_RANDOM_DEVICE_FILE_PATH_DEFAULT);
+        }
+        return devPath;
+    }
+
     /**
      * Gets path of native library.
      *
