diff --git a/src/main/java/org/apache/commons/crypto/random/OsCryptoRandom.java b/src/main/java/org/apache/commons/crypto/random/OsCryptoRandom.java
index a15686c..b93177a 100644
--- a/src/main/java/org/apache/commons/crypto/random/OsCryptoRandom.java
+++ b/src/main/java/org/apache/commons/crypto/random/OsCryptoRandom.java
@@ -24,7 +24,7 @@ import java.util.Properties;
 import java.util.Random;
 
 import org.apache.commons.crypto.conf.ConfigurationKeys;
-import org.apache.commons.crypto.utils.IoUtils;
+import org.apache.commons.crypto.utils.IOUtils;
 
 /**
  * A Random implementation that uses random bytes sourced from the operating
@@ -50,7 +50,7 @@ class OsCryptoRandom extends Random implements CryptoRandom {
     private void fillReservoir(int min) {
         if (pos >= reservoir.length - min) {
             try {
-                IoUtils.readFully(stream, reservoir, 0, reservoir.length);
+                IOUtils.readFully(stream, reservoir, 0, reservoir.length);
             } catch (IOException e) {
                 throw new RuntimeException("failed to fill reservoir", e);
             }
@@ -132,7 +132,7 @@ class OsCryptoRandom extends Random implements CryptoRandom {
     @Override
     synchronized public void close() {
         if (stream != null) {
-            IoUtils.cleanup(stream);
+            IOUtils.cleanup(stream);
             stream = null;
         }
     }
diff --git a/src/main/java/org/apache/commons/crypto/stream/PositionedCryptoInputStream.java b/src/main/java/org/apache/commons/crypto/stream/PositionedCryptoInputStream.java
index bd6c39d..d92e3aa 100644
--- a/src/main/java/org/apache/commons/crypto/stream/PositionedCryptoInputStream.java
+++ b/src/main/java/org/apache/commons/crypto/stream/PositionedCryptoInputStream.java
@@ -34,7 +34,7 @@ import javax.crypto.spec.IvParameterSpec;
 import org.apache.commons.crypto.cipher.CryptoCipher;
 import org.apache.commons.crypto.cipher.CryptoCipherFactory;
 import org.apache.commons.crypto.stream.input.Input;
-import org.apache.commons.crypto.utils.IoUtils;
+import org.apache.commons.crypto.utils.IOUtils;
 import org.apache.commons.crypto.utils.Utils;
 
 /**
@@ -133,7 +133,7 @@ public class PositionedCryptoInputStream extends CTRCryptoInputStream {
     public void readFully(long position, byte[] buffer, int offset, int length)
             throws IOException {
         checkStream();
-        IoUtils.readFully(input, position, buffer, offset, length);
+        IOUtils.readFully(input, position, buffer, offset, length);
         if (length > 0) {
             // This operation does not change the current offset of the file
             decrypt(position, buffer, offset, length);
diff --git a/src/main/java/org/apache/commons/crypto/utils/IoUtils.java b/src/main/java/org/apache/commons/crypto/utils/IOUtils.java
similarity index 96%
rename from src/main/java/org/apache/commons/crypto/utils/IoUtils.java
rename to src/main/java/org/apache/commons/crypto/utils/IOUtils.java
index 9c74db6..67f82ae 100644
--- a/src/main/java/org/apache/commons/crypto/utils/IoUtils.java
+++ b/src/main/java/org/apache/commons/crypto/utils/IOUtils.java
@@ -25,12 +25,12 @@ import org.apache.commons.crypto.stream.input.Input;
 /**
  * General utility methods for working with IO.
  */
-public final class IoUtils {
+public final class IOUtils {
 
     /**
-     * Teh private constructor of {@link IoUtils}.
+     * Teh private constructor of {@link IOUtils}.
      */
-    private IoUtils() {
+    private IOUtils() {
     }
 
     /**
