diff --git a/src/main/java/org/apache/commons/crypto/NativeCodeLoader.java b/src/main/java/org/apache/commons/crypto/NativeCodeLoader.java
index 8158908..1317431 100644
--- a/src/main/java/org/apache/commons/crypto/NativeCodeLoader.java
+++ b/src/main/java/org/apache/commons/crypto/NativeCodeLoader.java
@@ -88,12 +88,12 @@ final class NativeCodeLoader {
 
         // Load an OS-dependent native library inside a jar file
         nativeLibraryPath = "/org/apache/commons/crypto/native/"
-                + OsInfo.getNativeLibFolderPathForCurrentOS();
+                + OSInfo.getNativeLibFolderPathForCurrentOS();
         boolean hasNativeLib = hasResource(nativeLibraryPath + "/"
                 + nativeLibraryName);
         if (!hasNativeLib) {
             String altName = "libcommons-crypto.jnilib";
-            if (OsInfo.getOSName().equals("Mac") && hasResource(nativeLibraryPath + "/" + altName)) {
+            if (OSInfo.getOSName().equals("Mac") && hasResource(nativeLibraryPath + "/" + altName)) {
                 // Fix for openjdk7 for Mac
                 nativeLibraryName = altName;
                 hasNativeLib = true;
@@ -103,7 +103,7 @@ final class NativeCodeLoader {
         if (!hasNativeLib) {
             String errorMessage = String.format(
                     "no native library is found for os.name=%s and os.arch=%s",
-                    OsInfo.getOSName(), OsInfo.getArchName());
+                    OSInfo.getOSName(), OSInfo.getArchName());
             throw new RuntimeException(errorMessage);
         }
 
diff --git a/src/main/java/org/apache/commons/crypto/OsInfo.java b/src/main/java/org/apache/commons/crypto/OSInfo.java
similarity index 98%
rename from src/main/java/org/apache/commons/crypto/OsInfo.java
rename to src/main/java/org/apache/commons/crypto/OSInfo.java
index 4177022..cf386f7 100644
--- a/src/main/java/org/apache/commons/crypto/OsInfo.java
+++ b/src/main/java/org/apache/commons/crypto/OSInfo.java
@@ -24,7 +24,7 @@ import java.util.Locale;
 /**
  * Provides OS name and architecture name.
  */
-final class OsInfo {
+final class OSInfo {
     private final static HashMap<String, String> archMapping = new HashMap<>();
 
     /**
@@ -64,9 +64,9 @@ final class OsInfo {
     static final String PPC64 = "ppc64";
 
     /**
-     * The private constructor of {@link OsInfo}.
+     * The private constructor of {@link OSInfo}.
      */
-    private OsInfo() {
+    private OSInfo() {
     }
 
     static {
