diff --git a/src/main/java/org/apache/commons/crypto/stream/CtrCryptoInputStream.java b/src/main/java/org/apache/commons/crypto/stream/CTRCryptoInputStream.java
similarity index 93%
rename from src/main/java/org/apache/commons/crypto/stream/CtrCryptoInputStream.java
rename to src/main/java/org/apache/commons/crypto/stream/CTRCryptoInputStream.java
index 56ad048..6c6dacc 100644
--- a/src/main/java/org/apache/commons/crypto/stream/CtrCryptoInputStream.java
+++ b/src/main/java/org/apache/commons/crypto/stream/CTRCryptoInputStream.java
@@ -39,7 +39,7 @@ import org.apache.commons.crypto.utils.Utils;
 
 /**
  * <p>
- * CtrCryptoInputStream decrypts data. AES CTR mode is required in order to
+ * CTRCryptoInputStream decrypts data. AES CTR mode is required in order to
  * ensure that the plain text and cipher text have a 1:1 mapping. CTR crypto
  * stream has stream characteristic which is useful for implement features like
  * random seek. The decryption is buffer based. The key points of the decryption
@@ -51,7 +51,7 @@ import org.apache.commons.crypto.utils.Utils;
  * </p>
  * The underlying stream offset is maintained as state. It is not thread-safe.
  */
-public class CtrCryptoInputStream extends CryptoInputStream {
+public class CTRCryptoInputStream extends CryptoInputStream {
     /**
      * Underlying stream offset
      */
@@ -88,7 +88,7 @@ public class CtrCryptoInputStream extends CryptoInputStream {
     private static final int AES_BLOCK_SIZE = 16;
 
     /**
-     * Constructs a {@link CtrCryptoInputStream}.
+     * Constructs a {@link CTRCryptoInputStream}.
      *
      * @param props The <code>Properties</code> class represents a set of
      *        properties.
@@ -97,13 +97,13 @@ public class CtrCryptoInputStream extends CryptoInputStream {
      * @param iv Initialization vector for the cipher.
      * @throws IOException if an I/O error occurs.
      */
-    public CtrCryptoInputStream(Properties props, InputStream in, byte[] key,
+    public CTRCryptoInputStream(Properties props, InputStream in, byte[] key,
             byte[] iv) throws IOException {
         this(props, in, key, iv, 0);
     }
 
     /**
-     * Constructs a {@link CtrCryptoInputStream}.
+     * Constructs a {@link CTRCryptoInputStream}.
      *
      * @param props The <code>Properties</code> class represents a set of
      *        properties.
@@ -112,13 +112,13 @@ public class CtrCryptoInputStream extends CryptoInputStream {
      * @param iv Initialization vector for the cipher.
      * @throws IOException if an I/O error occurs.
      */
-    public CtrCryptoInputStream(Properties props, ReadableByteChannel in,
+    public CTRCryptoInputStream(Properties props, ReadableByteChannel in,
             byte[] key, byte[] iv) throws IOException {
         this(props, in, key, iv, 0);
     }
 
     /**
-     * Constructs a {@link CtrCryptoInputStream}.
+     * Constructs a {@link CTRCryptoInputStream}.
      *
      * @param in the input stream.
      * @param cipher the CryptoCipher instance.
@@ -127,13 +127,13 @@ public class CtrCryptoInputStream extends CryptoInputStream {
      * @param iv Initialization vector for the cipher.
      * @throws IOException if an I/O error occurs.
      */
-    protected CtrCryptoInputStream(InputStream in, CryptoCipher cipher,
+    protected CTRCryptoInputStream(InputStream in, CryptoCipher cipher,
             int bufferSize, byte[] key, byte[] iv) throws IOException {
         this(in, cipher, bufferSize, key, iv, 0);
     }
 
     /**
-     * Constructs a {@link CtrCryptoInputStream}.
+     * Constructs a {@link CTRCryptoInputStream}.
      *
      * @param in the ReadableByteChannel instance.
      * @param cipher the cipher instance.
@@ -142,13 +142,13 @@ public class CtrCryptoInputStream extends CryptoInputStream {
      * @param iv Initialization vector for the cipher.
      * @throws IOException if an I/O error occurs.
      */
-    protected CtrCryptoInputStream(ReadableByteChannel in, CryptoCipher cipher,
+    protected CTRCryptoInputStream(ReadableByteChannel in, CryptoCipher cipher,
             int bufferSize, byte[] key, byte[] iv) throws IOException {
         this(in, cipher, bufferSize, key, iv, 0);
     }
 
     /**
-     * Constructs a {@link CtrCryptoInputStream}.
+     * Constructs a {@link CTRCryptoInputStream}.
      *
      * @param input the input data.
      * @param cipher the CryptoCipher instance.
@@ -157,13 +157,13 @@ public class CtrCryptoInputStream extends CryptoInputStream {
      * @param iv Initialization vector for the cipher.
      * @throws IOException if an I/O error occurs.
      */
-    protected CtrCryptoInputStream(Input input, CryptoCipher cipher,
+    protected CTRCryptoInputStream(Input input, CryptoCipher cipher,
             int bufferSize, byte[] key, byte[] iv) throws IOException {
         this(input, cipher, bufferSize, key, iv, 0);
     }
 
     /**
-     * Constructs a {@link CtrCryptoInputStream}.
+     * Constructs a {@link CTRCryptoInputStream}.
      *
      * @param props The <code>Properties</code> class represents a set of
      *        properties.
@@ -173,7 +173,7 @@ public class CtrCryptoInputStream extends CryptoInputStream {
      * @param streamOffset the start offset in the stream.
      * @throws IOException if an I/O error occurs.
      */
-    public CtrCryptoInputStream(Properties props, InputStream in, byte[] key,
+    public CTRCryptoInputStream(Properties props, InputStream in, byte[] key,
             byte[] iv, long streamOffset) throws IOException {
         this(in, Utils.getCipherInstance(
                 "AES/CTR/NoPadding", props),
@@ -181,7 +181,7 @@ public class CtrCryptoInputStream extends CryptoInputStream {
     }
 
     /**
-     * Constructs a {@link CtrCryptoInputStream}.
+     * Constructs a {@link CTRCryptoInputStream}.
      *
      * @param props The <code>Properties</code> class represents a set of
      *        properties.
@@ -191,7 +191,7 @@ public class CtrCryptoInputStream extends CryptoInputStream {
      * @param streamOffset the start offset in the stream.
      * @throws IOException if an I/O error occurs.
      */
-    public CtrCryptoInputStream(Properties props, ReadableByteChannel in,
+    public CTRCryptoInputStream(Properties props, ReadableByteChannel in,
             byte[] key, byte[] iv, long streamOffset) throws IOException {
         this(in, Utils.getCipherInstance(
                 "AES/CTR/NoPadding", props),
@@ -199,7 +199,7 @@ public class CtrCryptoInputStream extends CryptoInputStream {
     }
 
     /**
-     * Constructs a {@link CtrCryptoInputStream}.
+     * Constructs a {@link CTRCryptoInputStream}.
      *
      * @param in the InputStream instance.
      * @param cipher the CryptoCipher instance.
@@ -209,7 +209,7 @@ public class CtrCryptoInputStream extends CryptoInputStream {
      * @param streamOffset the start offset in the stream.
      * @throws IOException if an I/O error occurs.
      */
-    protected CtrCryptoInputStream(InputStream in, CryptoCipher cipher,
+    protected CTRCryptoInputStream(InputStream in, CryptoCipher cipher,
             int bufferSize, byte[] key, byte[] iv, long streamOffset)
             throws IOException {
         this(new StreamInput(in, bufferSize), cipher, bufferSize, key, iv,
@@ -217,7 +217,7 @@ public class CtrCryptoInputStream extends CryptoInputStream {
     }
 
     /**
-     * Constructs a {@link CtrCryptoInputStream}.
+     * Constructs a {@link CTRCryptoInputStream}.
      *
      * @param in the ReadableByteChannel instance.
      * @param cipher the CryptoCipher instance.
@@ -227,14 +227,14 @@ public class CtrCryptoInputStream extends CryptoInputStream {
      * @param streamOffset the start offset in the stream.
      * @throws IOException if an I/O error occurs.
      */
-    protected CtrCryptoInputStream(ReadableByteChannel in, CryptoCipher cipher,
+    protected CTRCryptoInputStream(ReadableByteChannel in, CryptoCipher cipher,
             int bufferSize, byte[] key, byte[] iv, long streamOffset)
             throws IOException {
         this(new ChannelInput(in), cipher, bufferSize, key, iv, streamOffset);
     }
 
     /**
-     * Constructs a {@link CtrCryptoInputStream}.
+     * Constructs a {@link CTRCryptoInputStream}.
      *
      * @param input the input data.
      * @param cipher the CryptoCipher instance.
@@ -244,7 +244,7 @@ public class CtrCryptoInputStream extends CryptoInputStream {
      * @param streamOffset the start offset in the stream.
      * @throws IOException if an I/O error occurs.
      */
-    protected CtrCryptoInputStream(Input input, CryptoCipher cipher,
+    protected CTRCryptoInputStream(Input input, CryptoCipher cipher,
             int bufferSize, byte[] key, byte[] iv, long streamOffset)
             throws IOException {
         super(input, cipher, bufferSize, new SecretKeySpec(key, "AES"),
@@ -297,7 +297,7 @@ public class CtrCryptoInputStream extends CryptoInputStream {
     }
 
     /**
-     * Overrides the {@link CtrCryptoInputStream#read(ByteBuffer)}. Reads a
+     * Overrides the {@link CTRCryptoInputStream#read(ByteBuffer)}. Reads a
      * sequence of bytes from this channel into the given buffer.
      *
      * @param buf The buffer into which bytes are to be transferred.
@@ -556,7 +556,7 @@ public class CtrCryptoInputStream extends CryptoInputStream {
     }
 
     /**
-     * Overrides the {@link CtrCryptoInputStream#initCipher()}. Initializes the
+     * Overrides the {@link CTRCryptoInputStream#initCipher()}. Initializes the
      * cipher.
      */
     @Override
@@ -573,7 +573,7 @@ public class CtrCryptoInputStream extends CryptoInputStream {
      */
     protected void resetCipher(long position) throws IOException {
         final long counter = getCounter(position);
-        CtrCryptoInputStream.calculateIV(initIV, counter, iv);
+        CTRCryptoInputStream.calculateIV(initIV, counter, iv);
         try {
             cipher.init(Cipher.DECRYPT_MODE, key, new IvParameterSpec(iv));
         } catch (InvalidKeyException e) {
@@ -653,8 +653,8 @@ public class CtrCryptoInputStream extends CryptoInputStream {
      * @param IV the IV for input stream position
      */
     static void calculateIV(byte[] initIV, long counter, byte[] IV) {
-        Utils.checkArgument(initIV.length == CtrCryptoInputStream.AES_BLOCK_SIZE);
-        Utils.checkArgument(IV.length == CtrCryptoInputStream.AES_BLOCK_SIZE);
+        Utils.checkArgument(initIV.length == CTRCryptoInputStream.AES_BLOCK_SIZE);
+        Utils.checkArgument(IV.length == CTRCryptoInputStream.AES_BLOCK_SIZE);
     
         int i = IV.length; // IV length
         int j = 0; // counter bytes index
diff --git a/src/main/java/org/apache/commons/crypto/stream/CtrCryptoOutputStream.java b/src/main/java/org/apache/commons/crypto/stream/CTRCryptoOutputStream.java
similarity index 90%
rename from src/main/java/org/apache/commons/crypto/stream/CtrCryptoOutputStream.java
rename to src/main/java/org/apache/commons/crypto/stream/CTRCryptoOutputStream.java
index c05b0c0..6f9ea6b 100644
--- a/src/main/java/org/apache/commons/crypto/stream/CtrCryptoOutputStream.java
+++ b/src/main/java/org/apache/commons/crypto/stream/CTRCryptoOutputStream.java
@@ -39,7 +39,7 @@ import org.apache.commons.crypto.utils.Utils;
 
 /**
  * <p>
- * CtrCryptoOutputStream encrypts data. It is not thread-safe. AES CTR mode is
+ * CTRCryptoOutputStream encrypts data. It is not thread-safe. AES CTR mode is
  * required in order to ensure that the plain text and cipher text have a 1:1
  * mapping. The encryption is buffer based. The key points of the encryption are
  * (1) calculating counter and (2) padding through stream position.
@@ -50,7 +50,7 @@ import org.apache.commons.crypto.utils.Utils;
  * </p>
  * The underlying stream offset is maintained as state.
  */
-public class CtrCryptoOutputStream extends CryptoOutputStream {
+public class CTRCryptoOutputStream extends CryptoOutputStream {
     /**
      * Underlying stream offset.
      */
@@ -79,7 +79,7 @@ public class CtrCryptoOutputStream extends CryptoOutputStream {
     private boolean cipherReset = false;
 
     /**
-     * Constructs a {@link CtrCryptoOutputStream}.
+     * Constructs a {@link CTRCryptoOutputStream}.
      *
      * @param props The <code>Properties</code> class represents a set of
      *        properties.
@@ -88,13 +88,13 @@ public class CtrCryptoOutputStream extends CryptoOutputStream {
      * @param iv Initialization vector for the cipher.
      * @throws IOException if an I/O error occurs.
      */
-    public CtrCryptoOutputStream(Properties props, OutputStream out,
+    public CTRCryptoOutputStream(Properties props, OutputStream out,
             byte[] key, byte[] iv) throws IOException {
         this(props, out, key, iv, 0);
     }
 
     /**
-     * Constructs a {@link CtrCryptoOutputStream}.
+     * Constructs a {@link CTRCryptoOutputStream}.
      *
      * @param props The <code>Properties</code> class represents a set of
      *        properties.
@@ -103,13 +103,13 @@ public class CtrCryptoOutputStream extends CryptoOutputStream {
      * @param iv Initialization vector for the cipher.
      * @throws IOException if an I/O error occurs.
      */
-    public CtrCryptoOutputStream(Properties props, WritableByteChannel out,
+    public CTRCryptoOutputStream(Properties props, WritableByteChannel out,
             byte[] key, byte[] iv) throws IOException {
         this(props, out, key, iv, 0);
     }
 
     /**
-     * Constructs a {@link CtrCryptoOutputStream}.
+     * Constructs a {@link CTRCryptoOutputStream}.
      *
      * @param out the output stream.
      * @param cipher the CryptoCipher instance.
@@ -118,13 +118,13 @@ public class CtrCryptoOutputStream extends CryptoOutputStream {
      * @param iv Initialization vector for the cipher.
      * @throws IOException if an I/O error occurs.
      */
-    protected CtrCryptoOutputStream(OutputStream out, CryptoCipher cipher,
+    protected CTRCryptoOutputStream(OutputStream out, CryptoCipher cipher,
             int bufferSize, byte[] key, byte[] iv) throws IOException {
         this(out, cipher, bufferSize, key, iv, 0);
     }
 
     /**
-     * Constructs a {@link CtrCryptoOutputStream}.
+     * Constructs a {@link CTRCryptoOutputStream}.
      *
      * @param channel the WritableByteChannel instance.
      * @param cipher the CryptoCipher instance.
@@ -133,14 +133,14 @@ public class CtrCryptoOutputStream extends CryptoOutputStream {
      * @param iv Initialization vector for the cipher.
      * @throws IOException if an I/O error occurs.
      */
-    protected CtrCryptoOutputStream(WritableByteChannel channel,
+    protected CTRCryptoOutputStream(WritableByteChannel channel,
             CryptoCipher cipher, int bufferSize, byte[] key, byte[] iv)
             throws IOException {
         this(channel, cipher, bufferSize, key, iv, 0);
     }
 
     /**
-     * Constructs a {@link CtrCryptoOutputStream}.
+     * Constructs a {@link CTRCryptoOutputStream}.
      *
      * @param output the Output instance.
      * @param cipher the CryptoCipher instance.
@@ -149,13 +149,13 @@ public class CtrCryptoOutputStream extends CryptoOutputStream {
      * @param iv Initialization vector for the cipher.
      * @throws IOException if an I/O error occurs.
      */
-    protected CtrCryptoOutputStream(Output output, CryptoCipher cipher,
+    protected CTRCryptoOutputStream(Output output, CryptoCipher cipher,
             int bufferSize, byte[] key, byte[] iv) throws IOException {
         this(output, cipher, bufferSize, key, iv, 0);
     }
 
     /**
-     * Constructs a {@link CtrCryptoOutputStream}.
+     * Constructs a {@link CTRCryptoOutputStream}.
      *
      * @param props The <code>Properties</code> class represents a set of
      *        properties.
@@ -165,7 +165,7 @@ public class CtrCryptoOutputStream extends CryptoOutputStream {
      * @param streamOffset the start offset in the data.
      * @throws IOException if an I/O error occurs.
      */
-    public CtrCryptoOutputStream(Properties props, OutputStream out,
+    public CTRCryptoOutputStream(Properties props, OutputStream out,
             byte[] key, byte[] iv, long streamOffset) throws IOException {
         this(out, Utils.getCipherInstance(
                 "AES/CTR/NoPadding", props),
@@ -173,7 +173,7 @@ public class CtrCryptoOutputStream extends CryptoOutputStream {
     }
 
     /**
-     * Constructs a {@link CtrCryptoOutputStream}.
+     * Constructs a {@link CTRCryptoOutputStream}.
      *
      * @param props The <code>Properties</code> class represents a set of
      *        properties.
@@ -183,7 +183,7 @@ public class CtrCryptoOutputStream extends CryptoOutputStream {
      * @param streamOffset the start offset in the data.
      * @throws IOException if an I/O error occurs.
      */
-    public CtrCryptoOutputStream(Properties props, WritableByteChannel out,
+    public CTRCryptoOutputStream(Properties props, WritableByteChannel out,
             byte[] key, byte[] iv, long streamOffset) throws IOException {
         this(out, Utils.getCipherInstance(
                 "AES/CTR/NoPadding", props),
@@ -191,7 +191,7 @@ public class CtrCryptoOutputStream extends CryptoOutputStream {
     }
 
     /**
-     * Constructs a {@link CtrCryptoOutputStream}.
+     * Constructs a {@link CTRCryptoOutputStream}.
      *
      * @param out the output stream.
      * @param cipher the CryptoCipher instance.
@@ -201,7 +201,7 @@ public class CtrCryptoOutputStream extends CryptoOutputStream {
      * @param streamOffset the start offset in the data.
      * @throws IOException if an I/O error occurs.
      */
-    protected CtrCryptoOutputStream(OutputStream out, CryptoCipher cipher,
+    protected CTRCryptoOutputStream(OutputStream out, CryptoCipher cipher,
             int bufferSize, byte[] key, byte[] iv, long streamOffset)
             throws IOException {
         this(new StreamOutput(out, bufferSize), cipher, bufferSize, key, iv,
@@ -209,7 +209,7 @@ public class CtrCryptoOutputStream extends CryptoOutputStream {
     }
 
     /**
-     * Constructs a {@link CtrCryptoOutputStream}.
+     * Constructs a {@link CTRCryptoOutputStream}.
      *
      * @param channel the WritableByteChannel instance.
      * @param cipher the CryptoCipher instance.
@@ -219,7 +219,7 @@ public class CtrCryptoOutputStream extends CryptoOutputStream {
      * @param streamOffset the start offset in the data.
      * @throws IOException if an I/O error occurs.
      */
-    protected CtrCryptoOutputStream(WritableByteChannel channel,
+    protected CTRCryptoOutputStream(WritableByteChannel channel,
             CryptoCipher cipher, int bufferSize, byte[] key, byte[] iv,
             long streamOffset) throws IOException {
         this(new ChannelOutput(channel), cipher, bufferSize, key, iv,
@@ -227,7 +227,7 @@ public class CtrCryptoOutputStream extends CryptoOutputStream {
     }
 
     /**
-     * Constructs a {@link CtrCryptoOutputStream}.
+     * Constructs a {@link CTRCryptoOutputStream}.
      *
      * @param output the output stream.
      * @param cipher the CryptoCipher instance.
@@ -237,7 +237,7 @@ public class CtrCryptoOutputStream extends CryptoOutputStream {
      * @param streamOffset the start offset in the data.
      * @throws IOException if an I/O error occurs.
      */
-    protected CtrCryptoOutputStream(Output output, CryptoCipher cipher,
+    protected CTRCryptoOutputStream(Output output, CryptoCipher cipher,
             int bufferSize, byte[] key, byte[] iv, long streamOffset)
             throws IOException {
         super(output, cipher, bufferSize, new SecretKeySpec(key, "AES"),
@@ -325,7 +325,7 @@ public class CtrCryptoOutputStream extends CryptoOutputStream {
         padding = (byte) (streamOffset % cipher.getBlockSize());
         inBuffer.position(padding); // Set proper position for input data.
 
-        CtrCryptoInputStream.calculateIV(initIV, counter, iv);
+        CTRCryptoInputStream.calculateIV(initIV, counter, iv);
         try {
             cipher.init(Cipher.ENCRYPT_MODE, key, new IvParameterSpec(iv));
         } catch (InvalidKeyException e) {
diff --git a/src/main/java/org/apache/commons/crypto/stream/CryptoInputStream.java b/src/main/java/org/apache/commons/crypto/stream/CryptoInputStream.java
index 334d391..655f3b5 100644
--- a/src/main/java/org/apache/commons/crypto/stream/CryptoInputStream.java
+++ b/src/main/java/org/apache/commons/crypto/stream/CryptoInputStream.java
@@ -356,7 +356,7 @@ public class CryptoInputStream extends InputStream implements
     /**
      * Overrides the {@link InputStream#markSupported()}.
      *
-     * @return false,the {@link CtrCryptoInputStream} don't support the mark
+     * @return false,the {@link CTRCryptoInputStream} don't support the mark
      *         method.
      */
     @Override
diff --git a/src/main/java/org/apache/commons/crypto/stream/PositionedCryptoInputStream.java b/src/main/java/org/apache/commons/crypto/stream/PositionedCryptoInputStream.java
index b3d0509..bd6c39d 100644
--- a/src/main/java/org/apache/commons/crypto/stream/PositionedCryptoInputStream.java
+++ b/src/main/java/org/apache/commons/crypto/stream/PositionedCryptoInputStream.java
@@ -42,7 +42,7 @@ import org.apache.commons.crypto.utils.Utils;
  * starting at random position as well as provides the foundation for positioned
  * read for decrypting. This needs a stream cipher mode such as AES CTR mode.
  */
-public class PositionedCryptoInputStream extends CtrCryptoInputStream {
+public class PositionedCryptoInputStream extends CTRCryptoInputStream {
 
     /**
      * DirectBuffer pool
@@ -296,7 +296,7 @@ public class PositionedCryptoInputStream extends CtrCryptoInputStream {
     private void resetCipher(CipherState state, long position, byte[] iv)
             throws IOException {
         final long counter = getCounter(position);
-        CtrCryptoInputStream.calculateIV(getInitIV(), counter, iv);
+        CTRCryptoInputStream.calculateIV(getInitIV(), counter, iv);
         try {
             state.getCipher().init(Cipher.DECRYPT_MODE, key,
                     new IvParameterSpec(iv));
