diff --git a/src/test/java/org/apache/commons/crypto/stream/CbcNoPaddingCipherStreamTest.java b/src/test/java/org/apache/commons/crypto/stream/CBCNoPaddingCipherStreamTest.java
similarity index 94%
rename from src/test/java/org/apache/commons/crypto/stream/CbcNoPaddingCipherStreamTest.java
rename to src/test/java/org/apache/commons/crypto/stream/CBCNoPaddingCipherStreamTest.java
index d5a0de7..4fb21b6 100644
--- a/src/test/java/org/apache/commons/crypto/stream/CbcNoPaddingCipherStreamTest.java
+++ b/src/test/java/org/apache/commons/crypto/stream/CBCNoPaddingCipherStreamTest.java
@@ -19,7 +19,7 @@ package org.apache.commons.crypto.stream;
 
 import java.io.IOException;
 
-public class CbcNoPaddingCipherStreamTest extends AbstractCipherStreamTest {
+public class CBCNoPaddingCipherStreamTest extends AbstractCipherStreamTest {
 
     @Override
     public void setUp() throws IOException {
diff --git a/src/test/java/org/apache/commons/crypto/stream/CbcPkcs5PaddingCipherStreamTest.java b/src/test/java/org/apache/commons/crypto/stream/CBCPKCS5PaddingCipherStreamTest.java
similarity index 94%
rename from src/test/java/org/apache/commons/crypto/stream/CbcPkcs5PaddingCipherStreamTest.java
rename to src/test/java/org/apache/commons/crypto/stream/CBCPKCS5PaddingCipherStreamTest.java
index ebe3f60..99cd31f 100644
--- a/src/test/java/org/apache/commons/crypto/stream/CbcPkcs5PaddingCipherStreamTest.java
+++ b/src/test/java/org/apache/commons/crypto/stream/CBCPKCS5PaddingCipherStreamTest.java
@@ -19,7 +19,7 @@ package org.apache.commons.crypto.stream;
 
 import java.io.IOException;
 
-public class CbcPkcs5PaddingCipherStreamTest extends AbstractCipherStreamTest {
+public class CBCPKCS5PaddingCipherStreamTest extends AbstractCipherStreamTest {
 
     @Override
     public void setUp() throws IOException {
