diff --git a/src/test/java/org/apache/commons/crypto/cipher/AbstractCipherTest.java b/src/test/java/org/apache/commons/crypto/cipher/AbstractCipherTest.java
index 11908a1..9f5f3f3 100644
--- a/src/test/java/org/apache/commons/crypto/cipher/AbstractCipherTest.java
+++ b/src/test/java/org/apache/commons/crypto/cipher/AbstractCipherTest.java
@@ -31,7 +31,7 @@ import javax.crypto.spec.SecretKeySpec;
 import javax.xml.bind.DatatypeConverter;
 
 import org.apache.commons.crypto.conf.ConfigurationKeys;
-import org.apache.commons.crypto.jna.OpenSslJna;
+import org.apache.commons.crypto.jna.OpensslJna;
 import org.apache.commons.crypto.utils.ReflectionUtils;
 import org.apache.commons.crypto.utils.Utils;
 import org.junit.Assert;
@@ -44,7 +44,7 @@ public abstract class AbstractCipherTest {
 
     public static final String JCE_CIPHER_CLASSNAME = JceCipher.class.getName();
 
-    public static final String OPENSSLJNA_CIPHER_CLASSNAME = OpenSslJna.getCipherClass().getName();
+    public static final String OPENSSLJNA_CIPHER_CLASSNAME = OpensslJna.getCipherClass().getName();
 
     // data
     public static final int BYTEBUFFER_SIZE = 1000;
diff --git a/src/test/java/org/apache/commons/crypto/cipher/OpenSslCipherTest.java b/src/test/java/org/apache/commons/crypto/cipher/OpensslCipherTest.java
similarity index 99%
rename from src/test/java/org/apache/commons/crypto/cipher/OpenSslCipherTest.java
rename to src/test/java/org/apache/commons/crypto/cipher/OpensslCipherTest.java
index d144228..c5f0497 100644
--- a/src/test/java/org/apache/commons/crypto/cipher/OpenSslCipherTest.java
+++ b/src/test/java/org/apache/commons/crypto/cipher/OpensslCipherTest.java
@@ -29,7 +29,7 @@ import org.junit.Assert;
 import org.junit.Assume;
 import org.junit.Test;
 
-public class OpenSslCipherTest extends AbstractCipherTest {
+public class OpensslCipherTest extends AbstractCipherTest {
 
     @Override
     public void init() {
diff --git a/src/test/java/org/apache/commons/crypto/jna/OpenSslJnaCipherTest.java b/src/test/java/org/apache/commons/crypto/jna/OpensslJnaCipherTest.java
similarity index 95%
rename from src/test/java/org/apache/commons/crypto/jna/OpenSslJnaCipherTest.java
rename to src/test/java/org/apache/commons/crypto/jna/OpensslJnaCipherTest.java
index 6f5990a..30b7fde 100644
--- a/src/test/java/org/apache/commons/crypto/jna/OpenSslJnaCipherTest.java
+++ b/src/test/java/org/apache/commons/crypto/jna/OpensslJnaCipherTest.java
@@ -21,7 +21,7 @@ package org.apache.commons.crypto.jna;
 import org.apache.commons.crypto.cipher.AbstractCipherTest;
 import org.apache.commons.crypto.jna.OpenSslJnaCipher;
 
-public class OpenSslJnaCipherTest extends AbstractCipherTest {
+public class OpensslJnaCipherTest extends AbstractCipherTest {
 
     @Override
     public void init() {
diff --git a/src/test/java/org/apache/commons/crypto/jna/OpenSslJnaCryptoRandomTest.java b/src/test/java/org/apache/commons/crypto/jna/OpensslJnaCryptoRandomTest.java
similarity index 96%
rename from src/test/java/org/apache/commons/crypto/jna/OpenSslJnaCryptoRandomTest.java
rename to src/test/java/org/apache/commons/crypto/jna/OpensslJnaCryptoRandomTest.java
index 990bfd2..550b4a6 100644
--- a/src/test/java/org/apache/commons/crypto/jna/OpenSslJnaCryptoRandomTest.java
+++ b/src/test/java/org/apache/commons/crypto/jna/OpensslJnaCryptoRandomTest.java
@@ -28,7 +28,7 @@ import org.apache.commons.crypto.random.CryptoRandomFactory;
 
 import static org.junit.Assert.fail;
 
-public class OpenSslJnaCryptoRandomTest extends AbstractRandomTest {
+public class OpensslJnaCryptoRandomTest extends AbstractRandomTest {
 
     @Override
     public CryptoRandom getCryptoRandom() throws GeneralSecurityException {
