diff --git a/src/test/java/org/apache/commons/crypto/random/CryptoRandomFactoryTest.java b/src/test/java/org/apache/commons/crypto/random/CryptoRandomFactoryTest.java
index 2f78347..3f33519 100644
--- a/src/test/java/org/apache/commons/crypto/random/CryptoRandomFactoryTest.java
+++ b/src/test/java/org/apache/commons/crypto/random/CryptoRandomFactoryTest.java
@@ -44,7 +44,7 @@ public class CryptoRandomFactoryTest {
         Properties props = new Properties();
         CryptoRandom random = CryptoRandomFactory.getCryptoRandom(props);
         final String name = random.getClass().getName();
-        if (OpenSslCryptoRandom.isNativeCodeEnabled()) {
+        if (OpenSslCryptoRandom.isNativeCodeLoaded()) {
             Assert.assertEquals(OpenSslCryptoRandom.class.getName(), name);
         } else {
             Assert.assertEquals(JavaCryptoRandom.class.getName(), name);
