diff --git a/src/main/java/org/apache/commons/crypto/stream/CryptoOutputStream.java b/src/main/java/org/apache/commons/crypto/stream/CryptoOutputStream.java
index bb59e2f..efa1f95 100644
--- a/src/main/java/org/apache/commons/crypto/stream/CryptoOutputStream.java
+++ b/src/main/java/org/apache/commons/crypto/stream/CryptoOutputStream.java
@@ -360,9 +360,7 @@ public class CryptoOutputStream extends OutputStream implements
         outBuffer.flip();
 
         // write to output
-        while (outBuffer.hasRemaining()) {
-            output.write(outBuffer);
-        }
+        output.write(outBuffer);
     }
 
     /**
@@ -388,9 +386,7 @@ public class CryptoOutputStream extends OutputStream implements
         outBuffer.flip();
 
         // write to output
-        while (outBuffer.hasRemaining()) {
-            output.write(outBuffer);
-        }
+        output.write(outBuffer);
     }
 
     /**
