diff --git a/src/main/java/org/apache/commons/crypto/stream/CryptoInputStream.java b/src/main/java/org/apache/commons/crypto/stream/CryptoInputStream.java
index c3441fd..cd1e639 100644
--- a/src/main/java/org/apache/commons/crypto/stream/CryptoInputStream.java
+++ b/src/main/java/org/apache/commons/crypto/stream/CryptoInputStream.java
@@ -255,12 +255,8 @@ public class CryptoInputStream extends InputStream implements
             return n;
         }
         // No data in the out buffer, try read new data and decrypt it
-        // we loop for new data
-        int nd = 0;
-        while (nd == 0) {
-            nd = decryptMore();
-        }
-        if (nd < 0) {
+        int nd = decryptMore();
+        if (nd <= 0) {
             return nd;
         }
 
@@ -301,11 +297,7 @@ public class CryptoInputStream extends InputStream implements
             remaining -= outBuffer.remaining();
             outBuffer.clear();
 
-            // we loop for new data
-            nd = 0;
-            while (nd == 0) {
-                nd = decryptMore();
-            }
+            nd = decryptMore();
             if (nd < 0) {
                 break;
             }
@@ -410,12 +402,7 @@ public class CryptoInputStream extends InputStream implements
         int remaining = outBuffer.remaining();
         if (remaining <= 0) {
             // Decrypt more data
-            // we loop for new data
-            int nd = 0;
-            while (nd == 0) {
-                nd = decryptMore();
-            }
-
+            int nd = decryptMore();
             if (nd < 0) {
                 return -1;
             }
@@ -500,9 +487,7 @@ public class CryptoInputStream extends InputStream implements
      * will be put in the output buffer. If the end of the under stream reached,
      * we will do final of the cipher to finish all the decrypting of data.
      *
-     * @return The number of decrypted data.
-     *           return -1 (if end of the decrypted stream)
-     *           return 0 (no data now, but could have more later)
+     * @return The number of decrypted data. -1 if end of the decrypted stream.
      * @throws IOException if an I/O error occurs.
      */
     protected int decryptMore() throws IOException {
