diff --git a/src/main/java/org/apache/commons/crypto/stream/CryptoInputStream.java b/src/main/java/org/apache/commons/crypto/stream/CryptoInputStream.java
index 5c2bb14..c3441fd 100644
--- a/src/main/java/org/apache/commons/crypto/stream/CryptoInputStream.java
+++ b/src/main/java/org/apache/commons/crypto/stream/CryptoInputStream.java
@@ -58,16 +58,16 @@ public class CryptoInputStream extends InputStream implements
             + "stream.buffer.size";
 
     /** The CryptoCipher instance. */
-    protected final CryptoCipher cipher;
+    final CryptoCipher cipher;
 
     /** The buffer size. */
-    private final int bufferSize;
+    final int bufferSize;
 
     /** Crypto key for the cipher. */
-    protected final Key key;
+    final Key key;
 
     /** the algorithm parameters */
-    private final AlgorithmParameterSpec params;
+    final AlgorithmParameterSpec params;
 
     /** Flag to mark whether the input stream is closed. */
     private boolean closed;
@@ -78,19 +78,19 @@ public class CryptoInputStream extends InputStream implements
     private boolean finalDone = false;
 
     /** The input data. */
-    protected Input input;
+    Input input;
 
     /**
      * Input data buffer. The data starts at inBuffer.position() and ends at to
      * inBuffer.limit().
      */
-    protected ByteBuffer inBuffer;
+    ByteBuffer inBuffer;
 
     /**
      * The decrypted data buffer. The data starts at outBuffer.position() and
      * ends at outBuffer.limit().
      */
-    protected ByteBuffer outBuffer;
+    ByteBuffer outBuffer;
 
     // stream related configuration keys
     /**
diff --git a/src/main/java/org/apache/commons/crypto/stream/CryptoOutputStream.java b/src/main/java/org/apache/commons/crypto/stream/CryptoOutputStream.java
index 7e86809..bb59e2f 100644
--- a/src/main/java/org/apache/commons/crypto/stream/CryptoOutputStream.java
+++ b/src/main/java/org/apache/commons/crypto/stream/CryptoOutputStream.java
@@ -50,19 +50,19 @@ public class CryptoOutputStream extends OutputStream implements
     private final byte[] oneByteBuf = new byte[1];
 
     /** The output. */
-    protected Output output;
+    Output output;
 
     /** the CryptoCipher instance */
-    protected final CryptoCipher cipher;
+    final CryptoCipher cipher;
 
     /** The buffer size. */
-    private final int bufferSize;
+    final int bufferSize;
 
     /** Crypto key for the cipher. */
-    protected final Key key;
+    final Key key;
 
     /** the algorithm parameters */
-    private final AlgorithmParameterSpec params;
+    final AlgorithmParameterSpec params;
 
     /** Flag to mark whether the output stream is closed. */
     private boolean closed;
@@ -71,13 +71,13 @@ public class CryptoOutputStream extends OutputStream implements
      * Input data buffer. The data starts at inBuffer.position() and ends at
      * inBuffer.limit().
      */
-    protected ByteBuffer inBuffer;
+    ByteBuffer inBuffer;
 
     /**
      * Encrypted data buffer. The data starts at outBuffer.position() and ends
      * at outBuffer.limit().
      */
-    protected ByteBuffer outBuffer;
+    ByteBuffer outBuffer;
 
     /**
      * Constructs a {@link CryptoOutputStream}.
