diff --git a/src/main/java/org/apache/commons/crypto/stream/CTRCipherInputStream.java b/src/main/java/org/apache/commons/crypto/stream/CTRCipherInputStream.java
index 89f107e..f286ad8 100644
--- a/src/main/java/org/apache/commons/crypto/stream/CTRCipherInputStream.java
+++ b/src/main/java/org/apache/commons/crypto/stream/CTRCipherInputStream.java
@@ -375,7 +375,6 @@ public class CTRCipherInputStream extends CipherInputStream {
    * @return The number of decrypted data. -1 if end of the decrypted stream.
    * @throws IOException if an I/O error occurs.
    */
-  @Override
   protected int decryptMore() throws IOException {
     int n = input.read(inBuffer);
     if (n <= 0) {
@@ -395,7 +394,6 @@ public class CTRCipherInputStream extends CipherInputStream {
    *
    * @throws IOException if an I/O error occurs.
    */
-  @Override
   protected void decrypt() throws IOException {
     Utils.checkState(inBuffer.position() >= padding);
     if(inBuffer.position() == padding) {
diff --git a/src/main/java/org/apache/commons/crypto/stream/CipherOutputStream.java b/src/main/java/org/apache/commons/crypto/stream/CipherOutputStream.java
index 2418841..0106055 100644
--- a/src/main/java/org/apache/commons/crypto/stream/CipherOutputStream.java
+++ b/src/main/java/org/apache/commons/crypto/stream/CipherOutputStream.java
@@ -201,7 +201,6 @@ public class CipherOutputStream extends OutputStream implements
    * @param len the number of bytes to write.
    * @throws IOException if an I/O error occurs.
    */
-  @Override
   public void write(byte[] b, int off, int len) throws IOException {
     checkStream();
     if (b == null) {
