diff --git a/src/test/java/org/apache/commons/crypto/examples/CipherByteBufferExample.java b/src/test/java/org/apache/commons/crypto/examples/CipherByteBufferExample.java
index f35ba4c..c515461 100644
--- a/src/test/java/org/apache/commons/crypto/examples/CipherByteBufferExample.java
+++ b/src/test/java/org/apache/commons/crypto/examples/CipherByteBufferExample.java
@@ -99,7 +99,7 @@ public class CipherByteBufferExample {
      */
     private static String asString(ByteBuffer buffer) {
         final ByteBuffer copy = buffer.duplicate();
-        final byte[] bytes = new byte[copy.remaining()];
+        final byte[] bytes = new byte[Math.min(copy.remaining(),50)];
         copy.get(bytes);
         return new String(bytes, StandardCharsets.UTF_8);
     }
