diff --git a/src/main/java/org/apache/commons/crypto/stream/CryptoOutputStream.java b/src/main/java/org/apache/commons/crypto/stream/CryptoOutputStream.java
index b92d975..4246713 100644
--- a/src/main/java/org/apache/commons/crypto/stream/CryptoOutputStream.java
+++ b/src/main/java/org/apache/commons/crypto/stream/CryptoOutputStream.java
@@ -43,10 +43,6 @@ import org.apache.commons.crypto.utils.Utils;
  * {@link CryptoOutputStream} encrypts data and writes to the under layer
  * output. It supports any mode of operations such as AES CBC/CTR/GCM mode in
  * concept. It is not thread-safe.
- * <p>
- * This class should only be used with blocking sinks. Using this class to wrap
- * a non-blocking sink may lead to high CPU usage.
- * </p>
  */
 
 public class CryptoOutputStream extends OutputStream implements
diff --git a/src/main/java/org/apache/commons/crypto/stream/CtrCryptoOutputStream.java b/src/main/java/org/apache/commons/crypto/stream/CtrCryptoOutputStream.java
index 958d1a5..c05b0c0 100644
--- a/src/main/java/org/apache/commons/crypto/stream/CtrCryptoOutputStream.java
+++ b/src/main/java/org/apache/commons/crypto/stream/CtrCryptoOutputStream.java
@@ -48,13 +48,7 @@ import org.apache.commons.crypto.utils.Utils;
  * counter = base + pos/(algorithm blocksize); padding = pos%(algorithm
  * blocksize);
  * </p>
- * <p>
  * The underlying stream offset is maintained as state.
- * </p>
- * <p>
- * This class should only be used with blocking sinks. Using this class to wrap
- * a non-blocking sink may lead to high CPU usage.
- * </p>
  */
 public class CtrCryptoOutputStream extends CryptoOutputStream {
     /**
