diff --git a/src/test/java/org/apache/commons/crypto/cipher/OpenSslCipherTest.java b/src/test/java/org/apache/commons/crypto/cipher/OpenSslCipherTest.java
index 98fc325..de6d67c 100644
--- a/src/test/java/org/apache/commons/crypto/cipher/OpenSslCipherTest.java
+++ b/src/test/java/org/apache/commons/crypto/cipher/OpenSslCipherTest.java
@@ -22,11 +22,9 @@ import java.nio.ByteBuffer;
 import java.security.InvalidAlgorithmParameterException;
 import java.security.InvalidKeyException;
 import java.security.NoSuchAlgorithmException;
-import java.util.Properties;
 import javax.crypto.NoSuchPaddingException;
 import javax.crypto.ShortBufferException;
 import javax.crypto.spec.IvParameterSpec;
-import javax.crypto.spec.SecretKeySpec;
 
 import org.junit.Assert;
 import org.junit.Assume;
@@ -135,49 +133,4 @@ public class OpenSslCipherTest extends AbstractCipherTest {
         cipher.init(OpenSsl.ENCRYPT_MODE, KEY, new IvParameterSpec(invalidIV));
     }
 
-    @Test
-    public void testCipherLifecycle() throws Exception {
-        try (OpenSslCipher cipher = new OpenSslCipher(new Properties(), "AES/CTR/NoPadding")) {
-            try {
-                cipher.update(dummyBuffer(), dummyBuffer());
-                Assert.fail("Should have thrown exception.");
-            } catch (IllegalStateException ise) {
-                // expected;
-            }
-
-            cipher.init(OpenSsl.ENCRYPT_MODE, new SecretKeySpec(KEY, "AES"),
-                new IvParameterSpec(IV));
-            cipher.update(dummyBuffer(), dummyBuffer());
-
-            try {
-                cipher.init(OpenSsl.ENCRYPT_MODE, new SecretKeySpec(new byte[1], "AES"),
-                    new IvParameterSpec(IV));
-                Assert.fail("Should have thrown exception.");
-            } catch (InvalidKeyException ike) {
-                // expected;
-            }
-
-            // Should keep working with previous init parameters.
-            cipher.update(dummyBuffer(), dummyBuffer());
-            cipher.doFinal(dummyBuffer(), dummyBuffer());
-            cipher.close();
-
-            try {
-                cipher.update(dummyBuffer(), dummyBuffer());
-                Assert.fail("Should have thrown exception.");
-            } catch (IllegalStateException ise) {
-                // expected;
-            }
-
-            cipher.init(OpenSsl.ENCRYPT_MODE, new SecretKeySpec(KEY, "AES"),
-                new IvParameterSpec(IV));
-            cipher.update(dummyBuffer(), dummyBuffer());
-            cipher.close();
-        }
-    }
-
-    private ByteBuffer dummyBuffer() {
-        return ByteBuffer.allocateDirect(8);
-    }
-
 }
