diff --git a/src/test/java/org/apache/commons/crypto/stream/CBCNoPaddingCipherStreamTest.java b/src/test/java/org/apache/commons/crypto/stream/CBCNoPaddingCipherStreamTest.java
index 3b4c66e..c8717ff 100644
--- a/src/test/java/org/apache/commons/crypto/stream/CBCNoPaddingCipherStreamTest.java
+++ b/src/test/java/org/apache/commons/crypto/stream/CBCNoPaddingCipherStreamTest.java
@@ -23,8 +23,7 @@ import org.apache.commons.crypto.cipher.CipherTransformation;
 
 public class CBCNoPaddingCipherStreamTest extends AbstractCipherStreamTest {
 
-  @Override
-public void setUp() throws IOException {
+  public void setUp() throws IOException {
     transformation = CipherTransformation
         .AES_CBC_NOPADDING;
   }
diff --git a/src/test/java/org/apache/commons/crypto/stream/CBCPKCS5PaddingCipherStreamTest.java b/src/test/java/org/apache/commons/crypto/stream/CBCPKCS5PaddingCipherStreamTest.java
index c605a3b..ff34bc9 100644
--- a/src/test/java/org/apache/commons/crypto/stream/CBCPKCS5PaddingCipherStreamTest.java
+++ b/src/test/java/org/apache/commons/crypto/stream/CBCPKCS5PaddingCipherStreamTest.java
@@ -23,8 +23,7 @@ import org.apache.commons.crypto.cipher.CipherTransformation;
 
 public class CBCPKCS5PaddingCipherStreamTest extends AbstractCipherStreamTest {
 
-  @Override
-public void setUp() throws IOException {
+  public void setUp() throws IOException {
     transformation = CipherTransformation
         .AES_CBC_PKCS5PADDING;
   }
diff --git a/src/test/java/org/apache/commons/crypto/stream/CTRCipherStreamTest.java b/src/test/java/org/apache/commons/crypto/stream/CTRCipherStreamTest.java
index 6f9b5f1..f5e85ff 100644
--- a/src/test/java/org/apache/commons/crypto/stream/CTRCipherStreamTest.java
+++ b/src/test/java/org/apache/commons/crypto/stream/CTRCipherStreamTest.java
@@ -27,8 +27,7 @@ import org.apache.commons.crypto.cipher.CipherTransformation;
 
 public class CTRCipherStreamTest extends AbstractCipherStreamTest {
 
-  @Override
-public void setUp() throws IOException {
+  public void setUp() throws IOException {
     transformation = CipherTransformation
         .AES_CTR_NOPADDING;
   }
diff --git a/src/test/java/org/apache/commons/crypto/stream/CTRNoPaddingCipherStreamTest.java b/src/test/java/org/apache/commons/crypto/stream/CTRNoPaddingCipherStreamTest.java
index dc9907f..b59761d 100644
--- a/src/test/java/org/apache/commons/crypto/stream/CTRNoPaddingCipherStreamTest.java
+++ b/src/test/java/org/apache/commons/crypto/stream/CTRNoPaddingCipherStreamTest.java
@@ -23,8 +23,7 @@ import org.apache.commons.crypto.cipher.CipherTransformation;
 
 public class CTRNoPaddingCipherStreamTest extends AbstractCipherStreamTest {
 
-  @Override
-public void setUp() throws IOException {
+  public void setUp() throws IOException {
     transformation = CipherTransformation
         .AES_CTR_NOPADDING;
   }
